ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/monitor_sessions.py

```diff
@@ -0,0 +1,115 @@
+                    }
+                monitor_session["sources"].append(source)
+
+            if (session_settings := merged_settings.get("session_settings")) is not None:
+                monitor_session.update(session_settings)
+
+            monitor_sessions.append(monitor_session)
+
+        if monitor_sessions:
+            return strip_null_from_data(monitor_sessions, ([], {}, None))
+
+        return None
+
+    @cached_property
+    def _monitor_session_configs(self):
+        """
+        Return list of monitor session configs extracted from every interface
+        """
+        monitor_session_configs = []
+        for connected_endpoint in self._filtered_connected_endpoints:
+            for adapter in connected_endpoint["adapters"]:
+                if "monitor_sessions" not in adapter:
+                    continue
+
+                # Monitor session on Port-channel interface
+                if get(adapter, "port_channel.mode") is not None:
+                    default_channel_group_id = int("".join(re.findall(r"\d", adapter["switch_ports"][0])))
+                    channel_group_id = get(adapter, "port_channel.channel_id", default=default_channel_group_id)
+
+                    port_channel_interface_name = f"Port-Channel{channel_group_id}"
+                    for monitor_session in adapter["monitor_sessions"]:
+                        monitor_session["interface"] = port_channel_interface_name
+                        monitor_session_configs.append(monitor_session)
+                    continue
+
+                # Monitor session on Ethernet interface
+                for node_index, node_name in enumerate(adapter["switches"]):
+                    if node_name != self._hostname:
+                        continue
+
+                    for monitor_session in adapter["monitor_sessions"]:
+                        monitor_session["interface"] = adapter["switch_ports"][node_index]
+                        monitor_session_configs.append(monitor_session)
+
+        for network_port in self._filtered_network_ports:
+            if "monitor_sessions" not in network_port:
+                continue
+
+            for ethernet_interface_name in range_expand(network_port["switch_ports"]):
+                # Monitor session on Port-channel interface
+                if get(network_port, "port_channel.mode") is not None:
+                    default_channel_group_id = int("".join(re.findall(r"\d", ethernet_interface_name)))
+                    channel_group_id = get(network_port, "port_channel.channel_id", default=default_channel_group_id)
+
+                    port_channel_interface_name = f"Port-Channel{channel_group_id}"
+                    for monitor_session in adapter["monitor_sessions"]:
+                        monitor_session["interface"] = port_channel_interface_name
+                        monitor_session_configs.append(monitor_session)
+                    continue
```

Comment:
Could make a oneliner of these using `extend` + `map` + `lambda` though not sure it is worth it for readability

Reply:
You could say that about ~all our loops. Let's not go overboard.

Reply:
modified to extend with list comprehension and dict()